src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -164,6 +179,69 @@ private float getSentimentScore(String feedback) throws IOException {
     return queryList;
   }
 
+  private double getToxicityScore(String comment) throws IOException {
+    HttpURLConnection commentAnalyzerConnection = creatingPostRequestCommentAnalyzer();
+    // Create JSON request.
+    JSONObject jsonObjectCommentAnalyzer = jsonObjectCommentAnalyzerRequest(comment);
+
+    OutputStreamWriter writer = new OutputStreamWriter(commentAnalyzerConnection.getOutputStream());
+    writer.write(jsonObjectCommentAnalyzer.toString());
+    writer.close();
+
+    int responseCode =
+        commentAnalyzerConnection.getResponseCode(); // New items get NOT_FOUND on POST.
+    if (responseCode == HttpURLConnection.HTTP_OK
+        || responseCode == HttpURLConnection.HTTP_NOT_FOUND) {
+      StringBuilder response = new StringBuilder();
+      String line;
+
+      // Read input data stream.
+      BufferedReader reader =
+          new BufferedReader(new InputStreamReader(commentAnalyzerConnection.getInputStream()));
+      while ((line = reader.readLine()) != null) {
+        response.append(line);
+      }
+      reader.close();
+      JSONObject responseJsonObject = new JSONObject(response.toString());
+      double toxicityValue =
+          responseJsonObject
+              .getJSONObject("attributeScores")
+              .getJSONObject("TOXICITY")
+              .getJSONObject("summaryScore")
+              .getFloat("value");
+      return toxicityValue;
+    } else {
+      throw new RuntimeException("Cannot get toxicity score.");
+    }
+  }
+
+  private HttpURLConnection creatingPostRequestCommentAnalyzer() throws IOException {
+    String apiKey = "AIzaSyBnjF0OVUD3BGiuYFMSVe1_g134AKz3xQY";
+    URL urlCommentAnalyzer =
+        new URL("https://commentanalyzer.googleapis.com/v1alpha1/comments:analyze?key=" + apiKey);
+    HttpURLConnection connection = (HttpURLConnection) urlCommentAnalyzer.openConnection();
+    // Enable output for the connection.
+    connection.setDoOutput(true);
+    connection.setRequestProperty("Content-Type", "application/json; charset=UTF-8");
+    connection.setRequestProperty("Accept", "application/json");
+    connection.setReadTimeout(10000);
+    // Set HTTP request method.
+    connection.setRequestMethod("POST");
+    return connection;
+  }
+
+  private JSONObject jsonObjectCommentAnalyzerRequest(String text) throws IOException {
+    JSONObject jsonObject = new JSONObject();
+    JSONObject textJsonObject = new JSONObject();
+    JSONObject toxicityJsonObject = new JSONObject();
+    JSONObject toxicityScoreJsonObject = new JSONObject();
+    jsonObject.put("comment", textJsonObject.put("text", text));
+    jsonObject.put("languages", Arrays.asList("en"));
+    jsonObject.put(
+        "requestedAttributes", toxicityJsonObject.put("TOXICITY", toxicityScoreJsonObject));
```

Comment:
maybe make it "toxicity" so it's not styled like a constant

Reply:
you mean the all capitals "TOXICITY"? If so, it has to be like that because of the way the user requests the score. But I will definitely add a comment so it is less confusing.